server/client.py

```diff
@@ -55,25 +66,71 @@ def login_as(self, name, password):
         else:
             raise InvalidUserCredentials()
 
+    def recv(self, data):
+        """
+        Receive bytes and handle them
+        """
+        log.info("{self} got: {data}".format(self=self, data=data))
+        raw_messages = self.data_buffer.push(data)
+        # if we got complete message lets handle it
+        for raw_msg in raw_messages:
+            msg = self.data_parser.parse(raw_msg)
+            self.msg_dispatcher.dispatch(self, msg)
+
+    def handle_normal_msg(self, msg):
+        """
+        Handles NormalMessages
+        """
+        if msg.cmd == CMD_LOGIN:
+            if len(msg.params) != 2:
+                raise InvalidUserCredentials()
+            self.login_as(msg.params[0], msg.params[1])
+        if msg.cmd == CMD_LOGOUT:
+                self.logout()
+
+    def handle_payload_msg(self, msg):
+        """
+        Handles PayloadMessages
+        """
+        raise NotImplementedError()
+
+    def handle_error_msg(self, msg):
+        """
+        Handles ErrorMessages
+        """
+        raise NotImplementedError()
+
+    def handle_service_msg(self, msg):
+        """
+        Handles ServiceMessages
+        """
+        raise NotImplementedError()
+
     def send(self, msg):
         """
+        Sends Message or bytes to client
         @param msg: Message subclass instance or bytes or string
         """
         if type(msg) is str:
-            msg = msg.encode('utf-8')
+            msg = msg.encode('utf-8') + b'\n'
```

Comment:
Why should we add /n to the end of the message?

Reply:
It's a usability improvement for netcat users) it moves cursor to next line after response was printed